ts/client/src/client.ts

```diff
@@ -2083,6 +2086,52 @@ export class MangoClient {
     return await this.sendAndConfirmTransactionForGroup(group, [ix]);
   }
 
+  /**
+   * @param signature of the 'CONFIRMED' tx where perpPlaceOrder was called
+   * @returns list of orderIds, when null, then no order was placed on the book for that ix
+   */
+  public async perpPlaceOrderGetOrderIdsForTx(
+    signature: string,
+  ): Promise<(BN | null)[]> {
+    async function extractReturnValueForIx(
+      client: MangoClient,
+      ixName: string,
+    ): Promise<any[]> {
+      // we need a 'confirmed' level connection
+      const conn = new Connection(
+        client.program.provider.connection.rpcEndpoint,
+        'confirmed',
+      );
+      const tx = await conn.getTransaction(signature, {
+        maxSupportedTransactionVersion: 1,
+      });
+
+      const returnPrefix = `Program return: ${client.programId} `;
+      const returnLogs = tx?.meta?.logMessages?.filter((l) =>
+        l.startsWith(returnPrefix),
+      );
+      if (!returnLogs) {
+        throw new Error('Expected return log');
+      }
+
+      return returnLogs.map((returnLog) => {
```

Comment:
This should only be done for instructions that actually call the ixName instruction. Otherwise this would fail if another v4 instruction was called and also returned data, I think?